core/utils/dmljob.py

```diff
@@ -39,102 +32,231 @@ def __init__(
-    Returns:
-        DMLJob: A DML Job object from serialized_job.
+        Args:
+            hyperparams (dict): hyperparameters for training/validating a model.
+            raw_filepath (str): the filepath to the folder that contains the
+                                raw data before transform and training-test  
+                                split. Only one dataset file supported per
+                                folder at the moment.
+        """
+        self.job_type = JobTypes.JOB_SPLIT.name
+        self.hyperparams = hyperparams
+        self.raw_filepath = raw_filepath
+
+class DMLAverageJob(object):
     """
-    rest = serialized_job['job_data']
-    weights = None
-    if serialized_job.get('weights', None):
-        weights = deserialize_weights(serialized_job['weights'])
-    job = DMLJob(
-        rest['job_type'],
-        rest['serialized_model'],
-        rest['framework_type'],
+    DML Job for Averaging
+
+    Holds information specifically needed for running weighted average
+
+    """
+
+    def __init__(
+        self,
+        omega,
+        sigma_omega,
         weights,
-        rest['hyperparams'],
-        rest['label_column_name'],
-        rest['uuid'],
-        rest['raw_filepath'],
-        rest['session_filepath'],
-        rest['datapoint_count'],
-    )
-    job.omega = rest['omega']
-    job.sigma_omega = rest['sigma_omega']
-    return job
+        new_weights
+        ):
+        """
+        Initializes a DML Average Job object.
+
+        Args:
+            weights (list): list of np.arrays representing the weights of a model.
+            new_weights (list): list of np.arrays representing the weights that
+                                should be averaged with 'weights'
+            omega (float): the weight given to new_weights in weighted averaging
+            sigma_omega (float): the sum of omega's incorporated into weighted
+                                averaging so far. Kept in the DMLJob to allow the
+                                weighted average to be unrolled.
+        """
+        self.job_type = JobTypes.JOB_AVG.name
+        self.weights = weights
+        self.omega = omega
+        self.sigma_omega = sigma_omega
+        self.new_weights = new_weights
+
+class DMLJob(object):
+    """
+    DML Job
+
+    This class represents a generic job that the Scheduler and Runners can
+    execute. The job has some information that every job_type needs, but the
+    rest of the information is stored in type-specific job objects.
+
+    """
+
+    def __init__(
+        self,
+        job_type,
+        specific_job
+        ):
+        """
+        Initializes a DML Job object.
+
+        Args:
+            job_type (str): the job type.
+            specific_job (obj): the specific job container class
+        """
+        self.job_type = job_type
+        self.specific_job = specific_job
+    
+    def copy_constructor(self):
+        newjob = deepcopy(self)
+        return newjob
```

Comment:
I feel like this is a REALLY good use case for inheritance.

1. specific_job isn't really necessary anymore (just use `type(job)`)?
2. I think the only reason why having a DMLJob still here is so that Runner can verify that all jobs it s takes are DMLJobs. That might need to be refactored to isinstance, but I would argue it's better to not create any DMLJobs directly like you have above.
3. job_type might not actually be necessary anymore either, but that's up to you. Correct me if I'm wrong, but I think comparing class names basically replaces that functionality (i.e. checking if a job is X Job in tests, etc.)

I'd like to hear your thoughts.

Reply:
So you basically want all the specific DML Jobs to inherit from DMLJob, like DMLAverageJob(DMLJob)? Sure.

Reply:
Yes.

Reply:
Done

Reply:
Looks good! Feel free to merge